### ISSUE TITLE
Cannot read properties of undefined (reading 'replace')

### ISSUE THREAD

**Issue by kelson42** [openzim/mwoffliner#2046]
Tested with current dev version:

```
[warn] [2024-06-23T18:45:54.017Z] Failed to get CSS from [https://ar.wikipedia.org/w/load.php?lang=ar&modules=ext.gadget.Font-Amiri%2CFont-Arial%2CHideExLinkIcon%2CUrgent-fixes%2CWMP-icons%2CpalestineTheme&only=styles&skin=vector-2022]
[warn] [2024-06-23T18:45:54.020Z] Failed to get CSS from [https://ar.wikipedia.org/w/load.php?lang=ar&modules=site.styles&only=styles&skin=vector-2022]
[warn] [2024-06-23T18:45:54.020Z] Failed to get CSS from [https://ar.wikipedia.org/wiki/?title=Mediawiki%253Aoffline.css&action=raw]
[warn] [2024-06-23T18:45:54.042Z] Failed to get CSS from [https://ar.wikipedia.org/w/load.php?lang=ar&modules=ext.dismissableSiteNotice.styles%7Cext.flaggedRevs.basic%2Cicons%7Cext.inputBox.styles%7Cext.uls.interlanguage%7Cext.visualEditor.desktopArticleTarget.noscript%7Cext.wikimediaBadges%7Cext.wikimediamessages.styles%7Cmediawiki.ui.button%2Ccheckbox%2Cinput%7Cskins.vector.icons%2Cstyles%7Cskins.vector.search.codex.styles&only=styles&skin=vector-2022]
[error] [2024-06-23T18:45:54.243Z] Cannot read properties of undefined (reading 'replace')
[error] [2024-06-23T18:45:54.243Z] Error downloading article إليزابيث_ناصر
[error] [2024-06-23T18:45:54.244Z] Failed to run mwoffliner after [894s]: {
	"name": "Error",
	"message": "Cannot read properties of undefined (reading 'replace')"
}
[error] [2024-06-23T18:45:54.244Z] 

**********

Cannot read properties of undefined (reading 'replace')

**********



```

https://farm.openzim.org/pipeline/7c4f53d1-de0d-4297-bbc8-fe30b47b0a0a/debug

**Comment by kelson42** [comment]
Strongly suspect we have a bug which is a regression here,

**Comment by kelson42** [comment]
Same errormwith fi.wikiquote.org https://farm.openzim.org/pipeline/88665082-b26d-43c5-9426-fa86deec164c/debug

**Comment by kelson42** [comment]
Might be a duplicate of #1998

**Comment by audiodude** [comment]
This is caused by the fact that https://github.com/openzim/mwoffliner/blob/main/translation/fi.json is missing the `DISCLAIMER` field (https://github.com/openzim/mwoffliner/blob/main/translation/de.json#L8)

Not sure if we actually can't source this field and should code a fallback, or if we should be able to fill this in. Please advise @kelson42

**Comment by audiodude** [comment]
It's happening for `ar` because that has a `DISCLAIMER` but is missing `LICENSE_NAME` and `LAST_EDITED_ON`

**Comment by audiodude** [comment]
The code already has the concept of "fallback" language. Maybe instead of waiting for an exception/missing file, we should always just merge `translation/XX.json` with `translation/<fallback>.json` and use that?

**Comment by kelson42** [comment]
Fallback should aleays be "en". i don't understand why it works in general (fallbacking) but not here.... or it does not work in general?

**Comment by audiodude** [comment]
The "fallback" is "The entire language `xy` file is missing, so use the entire `en` file". It is not "The language `xy` file is missing the `DISCLAIMER` key, so use that key from `en`".

I have implemented the latter in #2050, PTAL.

**Comment by audiodude** [comment]
If we can't merge translation files, then the other option I think is to create a schema/typedef for the dump.strings object, and make sure that missing keys return empty string.
